// @thelicato/react-file-manager/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// lib/index.ts
var lib_exports = {};
__export(lib_exports, {
  ReactFileManager: () => ReactFileManager,
  ViewStyle: () => ViewStyle
});
module.exports = __toCommonJS(lib_exports);

// lib/ReactFileManager.tsx
var import_react12 = __toESM(require("react"), 1);

// lib/context/FileManagerContext.tsx
var import_react = require("react");
var FileManagerContext = (0, import_react.createContext)(null);
var useFileManager = () => {
  const context = (0, import_react.useContext)(FileManagerContext);
  if (!context) {
    throw new Error("useFileManager must be used within FileManagerProvider");
  }
  return context;
};

// lib/components/Navbar.tsx
var import_react2 = __toESM(require("react"), 1);
var Navbar = () => {
  const { fs, setCurrentFolder, onRefresh } = useFileManager();
  const initialFolders = (0, import_react2.useMemo)(() => {
    return fs.filter((f) => f.isDir && f.parentId === "0");
  }, [fs]);
  const handleClick = async (id) => {
    setCurrentFolder(id);
    if (onRefresh !== void 0) {
      try {
        await onRefresh(id);
      } catch (e) {
        throw new Error("Error during refresh");
      }
    }
  };
  return /* @__PURE__ */ import_react2.default.createElement("section", { className: "rfm-navbar" }, /* @__PURE__ */ import_react2.default.createElement(
    "span",
    {
      onClick: () => setCurrentFolder("0"),
      className: "rfm-navbar-root-link"
    },
    "Root"
  ), /* @__PURE__ */ import_react2.default.createElement("ul", { className: "rfm-navbar-list" }, initialFolders.map((f) => {
    return /* @__PURE__ */ import_react2.default.createElement(
      "li",
      {
        onClick: () => handleClick(f.id),
        className: "rfm-navbar-list-element",
        key: f.id
      },
      f.name
    );
  })));
};
var Navbar_default = Navbar;

// lib/components/Workspace.tsx
var import_react11 = __toESM(require("react"), 1);
var import_react_dropzone = require("react-dropzone");

// lib/types/Enums.ts
var ViewStyle = /* @__PURE__ */ ((ViewStyle2) => {
  ViewStyle2["List"] = "list";
  ViewStyle2["Icons"] = "icons";
  return ViewStyle2;
})(ViewStyle || {});

// lib/components/FileIcon.tsx
var import_react4 = __toESM(require("react"), 1);
var import_react5 = require("react");

// lib/components/SvgIcon.tsx
var import_react3 = __toESM(require("react"), 1);
var SvgIcon = ({
  svgType,
  ...props
}) => {
  const svgContent = () => {
    switch (svgType) {
      case "file": {
        return /* @__PURE__ */ import_react3.default.createElement(
          "svg",
          {
            version: "1",
            xmlns: "http://www.w3.org/2000/svg",
            viewBox: "0 0 48 48",
            enableBackground: "new 0 0 48 48"
          },
          /* @__PURE__ */ import_react3.default.createElement("polygon", { fill: "#90CAF9", points: "40,45 8,45 8,3 30,3 40,13" }),
          /* @__PURE__ */ import_react3.default.createElement("polygon", { fill: "#E1F5FE", points: "38.5,14 29,14 29,4.5" })
        );
      }
      case "folder": {
        return /* @__PURE__ */ import_react3.default.createElement(
          "svg",
          {
            version: "1",
            xmlns: "http://www.w3.org/2000/svg",
            viewBox: "0 0 48 48",
            enableBackground: "new 0 0 48 48"
          },
          /* @__PURE__ */ import_react3.default.createElement(
            "path",
            {
              fill: "#FFA000",
              d: "M40,12H22l-4-4H8c-2.2,0-4,1.8-4,4v8h40v-4C44,13.8,42.2,12,40,12z"
            }
          ),
          /* @__PURE__ */ import_react3.default.createElement(
            "path",
            {
              fill: "#FFCA28",
              d: "M40,12H8c-2.2,0-4,1.8-4,4v20c0,2.2,1.8,4,4,4h32c2.2,0,4-1.8,4-4V16C44,13.8,42.2,12,40,12z"
            }
          )
        );
      }
      case "arrow-up": {
        return /* @__PURE__ */ import_react3.default.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", fill: "currentColor", viewBox: "0 0 1024 1024" }, /* @__PURE__ */ import_react3.default.createElement("path", { d: "M868 545.5L536.1 163a31.96 31.96 0 0 0-48.3 0L156 545.5a7.97 7.97 0 0 0 6 13.2h81c4.6 0 9-2 12.1-5.5L474 300.9V864c0 4.4 3.6 8 8 8h60c4.4 0 8-3.6 8-8V300.9l218.9 252.3c3 3.5 7.4 5.5 12.1 5.5h81c6.8 0 10.5-8 6-13.2z" }));
      }
      case "arrow-down": {
        return /* @__PURE__ */ import_react3.default.createElement("svg", { stroke: "currentColor", fill: "currentColor", strokeWidth: "0", viewBox: "0 0 1024 1024", xmlns: "http://www.w3.org/2000/svg" }, /* @__PURE__ */ import_react3.default.createElement("path", { d: "M862 465.3h-81c-4.6 0-9 2-12.1 5.5L550 723.1V160c0-4.4-3.6-8-8-8h-60c-4.4 0-8 3.6-8 8v563.1L255.1 470.8c-3-3.5-7.4-5.5-12.1-5.5h-81c-6.8 0-10.5 8.1-6 13.2L487.9 861a31.96 31.96 0 0 0 48.3 0L868 478.5c4.5-5.2.8-13.2-6-13.2z" }));
      }
      case "close": {
        return /* @__PURE__ */ import_react3.default.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 1024 1024" }, /* @__PURE__ */ import_react3.default.createElement("path", { d: "M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z" }));
      }
      case "list": {
        return /* @__PURE__ */ import_react3.default.createElement("svg", { stroke: "currentColor", fill: "currentColor", strokeWidth: "0", viewBox: "0 0 256 256", xmlns: "http://www.w3.org/2000/svg" }, /* @__PURE__ */ import_react3.default.createElement("path", { d: "M80,64a8,8,0,0,1,8-8H216a8,8,0,0,1,0,16H88A8,8,0,0,1,80,64Zm136,56H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16Zm0,64H88a8,8,0,0,0,0,16H216a8,8,0,0,0,0-16ZM44,52A12,12,0,1,0,56,64,12,12,0,0,0,44,52Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,116Zm0,64a12,12,0,1,0,12,12A12,12,0,0,0,44,180Z" }));
      }
      case "icons": {
        return /* @__PURE__ */ import_react3.default.createElement("svg", { stroke: "currentColor", fill: "currentColor", strokeWidth: "0", viewBox: "0 0 256 256", xmlns: "http://www.w3.org/2000/svg" }, /* @__PURE__ */ import_react3.default.createElement("path", { d: "M72,60A12,12,0,1,1,60,48,12,12,0,0,1,72,60Zm56-12a12,12,0,1,0,12,12A12,12,0,0,0,128,48Zm68,24a12,12,0,1,0-12-12A12,12,0,0,0,196,72ZM60,116a12,12,0,1,0,12,12A12,12,0,0,0,60,116Zm68,0a12,12,0,1,0,12,12A12,12,0,0,0,128,116Zm68,0a12,12,0,1,0,12,12A12,12,0,0,0,196,116ZM60,184a12,12,0,1,0,12,12A12,12,0,0,0,60,184Zm68,0a12,12,0,1,0,12,12A12,12,0,0,0,128,184Zm68,0a12,12,0,1,0,12,12A12,12,0,0,0,196,184Z" }));
      }
      default: {
        return "";
      }
    }
  };
  return /* @__PURE__ */ import_react3.default.createElement("div", { ...props }, svgContent());
};
var SvgIcon_default = SvgIcon;

// lib/components/FileIcon.tsx
var FileIcon = (props) => {
  const { setCurrentFolder, onRefresh } = useFileManager();
  const handleClick = async () => {
    if (props.isDir) {
      setCurrentFolder(props.id);
      if (onRefresh !== void 0) {
        try {
          await onRefresh(props.id);
        } catch (e) {
          throw new Error("Error during refresh");
        }
      }
    }
  };
  const fileExtension = (0, import_react5.useMemo)(() => {
    if (!props.name.includes(".")) {
      return "";
    }
    const nameArray = props.name.split(".");
    return `.${nameArray[nameArray.length - 1]}`;
  }, [props.id]);
  return /* @__PURE__ */ import_react4.default.createElement(import_react4.default.Fragment, null, /* @__PURE__ */ import_react4.default.createElement("div", { onClick: handleClick, className: "rfm-file-icon-container" }, /* @__PURE__ */ import_react4.default.createElement(
    SvgIcon_default,
    {
      svgType: props.isDir ? "folder" : "file",
      className: "rfm-file-icon-svg"
    }
  ), /* @__PURE__ */ import_react4.default.createElement("span", { className: "rfm-file-icon-extension" }, fileExtension), /* @__PURE__ */ import_react4.default.createElement("span", { className: "rfm-file-icon-name" }, props.name)));
};
var FileIcon_default = FileIcon;

// lib/components/NewFolderIcon.tsx
var import_react6 = __toESM(require("react"), 1);
var NewFolderIcon = ({
  onClick
}) => {
  return /* @__PURE__ */ import_react6.default.createElement("div", { onClick, className: "rfm-folder-icon-container" }, /* @__PURE__ */ import_react6.default.createElement("span", { className: "rfm-folder-icon-span" }, "+"));
};
var NewFolderIcon_default = NewFolderIcon;

// lib/components/FolderPath.tsx
var import_react7 = __toESM(require("react"), 1);
var FolderPath = () => {
  const { fs, currentFolder, setCurrentFolder, viewStyle, setViewStyle } = useFileManager();
  const goUp = () => {
    const currentFolderInfo = fs.find((f) => f.id === currentFolder);
    if (currentFolderInfo && currentFolderInfo.parentId) {
      setCurrentFolder(currentFolderInfo.parentId);
    }
  };
  const parentPath = (0, import_react7.useMemo)(() => {
    const parentId = fs.find(
      (f) => f.id === currentFolder
    )?.parentId;
    if (!parentId) {
      return "";
    }
    const parentDir = fs.find((f) => f.id === parentId);
    if (!parentDir?.path) {
      return "";
    }
    const _parentPath = parentDir.path.slice(-1) === "/" ? parentDir.path : `${parentDir.path}/`;
    return _parentPath;
  }, [fs, currentFolder]);
  const currentPath = (0, import_react7.useMemo)(() => {
    const currentFolderInfo = fs.find((f) => f.id === currentFolder);
    return currentFolderInfo ? currentFolderInfo.name : "";
  }, [fs, currentFolder]);
  return /* @__PURE__ */ import_react7.default.createElement("div", { className: "rfm-workspace-header" }, /* @__PURE__ */ import_react7.default.createElement("div", { className: "rfm-folder-path-container" }, /* @__PURE__ */ import_react7.default.createElement(
    SvgIcon_default,
    {
      svgType: "arrow-up",
      onClick: goUp,
      className: "rfm-folder-path-svg"
    }
  ), /* @__PURE__ */ import_react7.default.createElement("span", { className: "rfm-folder-path-span" }, parentPath, /* @__PURE__ */ import_react7.default.createElement("b", null, currentPath))), /* @__PURE__ */ import_react7.default.createElement("div", { className: "rfm-header-container" }, /* @__PURE__ */ import_react7.default.createElement(
    SvgIcon_default,
    {
      svgType: "list",
      className: `rfm-header-icon ${viewStyle === "list" /* List */ && "rfm-header-icon--selected"}`,
      onClick: () => setViewStyle("list" /* List */)
    }
  ), /* @__PURE__ */ import_react7.default.createElement(
    SvgIcon_default,
    {
      svgType: "icons",
      className: `rfm-header-icon ${viewStyle === "icons" /* Icons */ && "rfm-header-icon--selected"}`,
      onClick: () => setViewStyle("icons" /* Icons */)
    }
  )));
};
var FolderPath_default = FolderPath;

// lib/components/NewFolderModal.tsx
var import_react9 = __toESM(require("react"), 1);

// lib/components/CommonModal.tsx
var import_react8 = __toESM(require("react"), 1);
var import_react_draggable = __toESM(require("react-draggable"), 1);
var CommonModal = ({
  children,
  title,
  isVisible,
  onClose
}) => {
  if (!isVisible) {
    return /* @__PURE__ */ import_react8.default.createElement(import_react8.default.Fragment, null);
  }
  return /* @__PURE__ */ import_react8.default.createElement(import_react_draggable.default, { bounds: "#react-file-manager-workspace" }, /* @__PURE__ */ import_react8.default.createElement("div", { className: "rfm-modal-container" }, /* @__PURE__ */ import_react8.default.createElement("div", null, /* @__PURE__ */ import_react8.default.createElement("h3", { className: "rfm-modal-title" }, title), /* @__PURE__ */ import_react8.default.createElement(
    SvgIcon_default,
    {
      onClick: onClose,
      svgType: "close",
      className: "rfm-modal-icon"
    }
  )), children));
};
var CommonModal_default = CommonModal;

// lib/components/NewFolderModal.tsx
var NewFolderModal = (props) => {
  const { onCreateFolder } = useFileManager();
  const folderName = (0, import_react9.useRef)();
  const onConfirm = async () => {
    if (folderName && folderName.current && folderName.current.value && folderName.current.value.length > 0 && onCreateFolder) {
      await onCreateFolder(folderName.current.value);
    }
  };
  return /* @__PURE__ */ import_react9.default.createElement(CommonModal_default, { title: "Create New Folder", ...props }, /* @__PURE__ */ import_react9.default.createElement("div", null, /* @__PURE__ */ import_react9.default.createElement("form", { className: "rfm-new-folder-modal-form" }, /* @__PURE__ */ import_react9.default.createElement("div", null, /* @__PURE__ */ import_react9.default.createElement(
    "input",
    {
      ref: folderName,
      type: "text",
      className: "rfm-new-folder-modal-input",
      placeholder: "Folder Name",
      required: true
    }
  )), /* @__PURE__ */ import_react9.default.createElement(
    "button",
    {
      onClick: onConfirm,
      disabled: folderName && folderName.current && folderName.current.value && folderName.current.value.length === 0,
      type: "submit",
      className: "rfm-new-folder-modal-btn"
    },
    "Create"
  ))));
};
var NewFolderModal_default = NewFolderModal;

// lib/components/UploadFileModal.tsx
var import_react10 = __toESM(require("react"), 1);
var UploadFileModal = (props) => {
  const { onUpload, uploadedFileData, currentFolder } = useFileManager();
  const onConfirm = async () => {
    if (onUpload && uploadedFileData) {
      await onUpload(uploadedFileData, currentFolder);
    }
  };
  return /* @__PURE__ */ import_react10.default.createElement(CommonModal_default, { title: "Upload file", ...props }, /* @__PURE__ */ import_react10.default.createElement("div", null, /* @__PURE__ */ import_react10.default.createElement("h4", { className: "rfm-upload-file-modal-title" }, "Are you sure you want to upload the file?"), /* @__PURE__ */ import_react10.default.createElement("div", { className: "rfm-upload-file-modal-container" }, /* @__PURE__ */ import_react10.default.createElement(
    "button",
    {
      onClick: onConfirm,
      type: "submit",
      className: "rfm-upload-file-modal-btn rfm-upload-file-modal-btn-confirm"
    },
    "Upload"
  ), /* @__PURE__ */ import_react10.default.createElement(
    "button",
    {
      onClick: props.onClose,
      type: "submit",
      className: "rfm-upload-file-modal-btn rfm-upload-file-modal-btn-cancel"
    },
    "Cancel"
  ))));
};
var UploadFileModal_default = UploadFileModal;

// lib/components/Workspace.tsx
var import_react_table = require("@tanstack/react-table");
var columnHelper = (0, import_react_table.createColumnHelper)();
var columns = [
  columnHelper.accessor("name", {
    header: () => "Name",
    cell: (info) => /* @__PURE__ */ import_react11.default.createElement("div", { className: "rfm-workspace-list-icon-td" }, /* @__PURE__ */ import_react11.default.createElement(SvgIcon_default, { svgType: info.row.original.isDir ? "folder" : "file", className: "rfm-workspace-list-icon" }), /* @__PURE__ */ import_react11.default.createElement("p", null, info.getValue()))
  }),
  columnHelper.accessor("lastModified", {
    header: () => "Last Modified",
    cell: (info) => info.getValue() ? new Date(info.getValue() * 1e3).toLocaleString() : "N/A"
  })
];
var Workspace = () => {
  const { currentFolder, fs, viewStyle, viewOnly, setCurrentFolder, setUploadedFileData, onDoubleClick, onRefresh } = useFileManager();
  const [newFolderModalVisible, setNewFolderModalVisible] = (0, import_react11.useState)(false);
  const [uploadFileModalVisible, setUploadFileModalVisible] = (0, import_react11.useState)(false);
  const setUploadModalVisible = (value) => {
    if (viewOnly) {
      setUploadFileModalVisible(false);
    } else {
      setUploadFileModalVisible(value);
    }
  };
  (0, import_react11.useEffect)(() => {
    if (newFolderModalVisible) {
      setNewFolderModalVisible(false);
    }
    if (uploadFileModalVisible) {
      setUploadModalVisible(false);
      setUploadedFileData(void 0);
    }
  }, [currentFolder]);
  const onDrop = (0, import_react11.useCallback)(
    (acceptedFiles) => {
      const file = acceptedFiles[0];
      setUploadedFileData(file);
      setUploadModalVisible(true);
    },
    [setUploadedFileData]
  );
  const onCloseUploadFileModal = () => {
    setUploadModalVisible(false);
    setUploadedFileData(void 0);
  };
  const { getRootProps, isDragAccept } = (0, import_react_dropzone.useDropzone)({
    noClick: true,
    noKeyboard: true,
    onDrop
  });
  const currentFolderFiles = (0, import_react11.useMemo)(() => {
    const files = fs.filter((f) => f.parentId === currentFolder);
    return files;
  }, [fs, currentFolder]);
  const table = (0, import_react_table.useReactTable)({
    data: currentFolderFiles,
    columns,
    getCoreRowModel: (0, import_react_table.getCoreRowModel)(),
    getSortedRowModel: (0, import_react_table.getSortedRowModel)(),
    initialState: {
      sorting: [{ id: "name", desc: false }]
    }
  });
  const handleClick = async (file) => {
    if (file.isDir) {
      setCurrentFolder(file.id);
      if (onRefresh !== void 0) {
        try {
          await onRefresh(file.id);
        } catch (e) {
          throw new Error("Error during refresh");
        }
      }
    }
  };
  const handleDoubleClick = (id) => {
    if (onDoubleClick) {
      onDoubleClick(id);
    }
  };
  return /* @__PURE__ */ import_react11.default.createElement(
    "section",
    {
      id: "react-file-manager-workspace",
      className: `rfm-workspace ${isDragAccept && !viewOnly ? "rfm-workspace-dropzone" : ""}`,
      ...getRootProps()
    },
    /* @__PURE__ */ import_react11.default.createElement(FolderPath_default, null),
    /* @__PURE__ */ import_react11.default.createElement("div", { className: "rfm-workspace-file-listing" }, viewStyle === "icons" /* Icons */ && /* @__PURE__ */ import_react11.default.createElement(import_react11.default.Fragment, null, currentFolderFiles.map(
      (f, key) => {
        return /* @__PURE__ */ import_react11.default.createElement("button", { onDoubleClick: () => handleDoubleClick(f.id), key }, /* @__PURE__ */ import_react11.default.createElement(FileIcon_default, { id: f.id, name: f.name, isDir: f.isDir }));
      }
    ), !viewOnly && /* @__PURE__ */ import_react11.default.createElement(NewFolderIcon_default, { onClick: () => setNewFolderModalVisible(true) })), viewStyle === "list" /* List */ && /* @__PURE__ */ import_react11.default.createElement(import_react11.default.Fragment, null, /* @__PURE__ */ import_react11.default.createElement("table", { className: "w-full" }, /* @__PURE__ */ import_react11.default.createElement("thead", null, table.getHeaderGroups().map((headerGroup) => /* @__PURE__ */ import_react11.default.createElement("tr", { key: headerGroup.id }, headerGroup.headers.map((header) => /* @__PURE__ */ import_react11.default.createElement("th", { className: "rfm-workspace-list-th", key: header.id, onClick: header.column.getToggleSortingHandler() }, /* @__PURE__ */ import_react11.default.createElement("div", { className: "rfm-workspace-list-th-content" }, (0, import_react_table.flexRender)(header.column.columnDef.header, header.getContext()), header.column.getIsSorted() ? header.column.getIsSorted() === "desc" ? /* @__PURE__ */ import_react11.default.createElement(SvgIcon_default, { svgType: "arrow-down", className: "rfm-header-sort-icon" }) : /* @__PURE__ */ import_react11.default.createElement(SvgIcon_default, { svgType: "arrow-up", className: "rfm-header-sort-icon" }) : "")))))), /* @__PURE__ */ import_react11.default.createElement("tbody", null, table.getRowModel().rows.map((row) => /* @__PURE__ */ import_react11.default.createElement("tr", { key: row.id, className: "rfm-workspace-list-icon-row" }, row.getVisibleCells().map((cell) => /* @__PURE__ */ import_react11.default.createElement("td", { className: "rfm-workspace-list-align-txt", key: cell.id, onClick: () => handleClick(row.original), onDoubleClick: () => handleDoubleClick(row.original.id) }, (0, import_react_table.flexRender)(cell.column.columnDef.cell, cell.getContext()))))))), !viewOnly && /* @__PURE__ */ import_react11.default.createElement("button", { className: "rfm-workspace-list-add-folder", onClick: () => setNewFolderModalVisible(true) }, "Add Folder")), !viewOnly && /* @__PURE__ */ import_react11.default.createElement(import_react11.default.Fragment, null, /* @__PURE__ */ import_react11.default.createElement(
      NewFolderModal_default,
      {
        isVisible: newFolderModalVisible,
        onClose: () => setNewFolderModalVisible(false)
      }
    ), /* @__PURE__ */ import_react11.default.createElement(
      UploadFileModal_default,
      {
        isVisible: uploadFileModalVisible,
        onClose: onCloseUploadFileModal
      }
    )))
  );
};
var Workspace_default = Workspace;

// lib/ReactFileManager.tsx
var ReactFileManager = ({
  fs,
  viewOnly,
  onDoubleClick,
  onRefresh,
  onUpload,
  onCreateFolder,
  onDelete
}) => {
  const [currentFolder, setCurrentFolder] = (0, import_react12.useState)("0");
  const [uploadedFileData, setUploadedFileData] = (0, import_react12.useState)();
  const [viewStyle, setViewStyle] = (0, import_react12.useState)("list" /* List */);
  return /* @__PURE__ */ import_react12.default.createElement(
    FileManagerContext.Provider,
    {
      value: {
        fs,
        viewStyle,
        setViewStyle,
        viewOnly,
        currentFolder,
        setCurrentFolder,
        onDoubleClick,
        onRefresh,
        onUpload,
        onCreateFolder,
        onDelete,
        uploadedFileData,
        setUploadedFileData
      }
    },
    /* @__PURE__ */ import_react12.default.createElement("div", { className: "rfm-main-container" }, /* @__PURE__ */ import_react12.default.createElement(Navbar_default, null), /* @__PURE__ */ import_react12.default.createElement(Workspace_default, null))
  );
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ReactFileManager,
  ViewStyle
});
